### DISCUSSION TITLE
Need help with local Jekyll

### DISCUSSION THREAD

**Discussion by berndbausch** [community/community#21914] [Other Feature Feedback, Questions, & Ideas]
<p>In the following, you will see that I am a git newbie who doesn’t know what he’s doing. Sorry for any overly naive questions.</p>
<p>I have a few Github pages already. So far, I have managed them directly on the web site, but would like to try managing them locally and using Jekyll. </p>
<p>When I follow <a href="https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/#platform-linux" rel="nofollow noopener">https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/#platform-linux</a>, I reach a roadblock in step 3, “generate Jekyll site files”. </p>
<p><strong>What am I supposed to do</strong> between steps 3-6, “connect your remote repository”, and 3-7, “edit the Jekyll template site”? I tried <em>git fetch</em>, which succeeded. How do I now get my files into the working tree? <em>git checkout master</em> complains that it would overwrite <em>.gitignore</em> and <em>_config.yml</em>. <em>git merge origin/master</em> complains that it would overwrite <em>.gitignore</em>. Any suggestions what I should do, or what I should read to continue the process?</p>
<p>This is on Fedora 23, by the way.</p>
<p>Since I think the referenced help page has room for improvement, I would like to comment on it. See also </p>
<p><a href="https://github.community/t/how-can-i-provide-feedback-to-githubs-help/643">https://github.community/t5/How-to-use-Git-and-GitHub/How-can-I-provide-feedback-to-Github-s-help/td-p/4066/jump-to/first-unread-message</a></p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Since you started out with your Pages site on GitHub, did you <a href="https://help.github.com/articles/cloning-a-repository/" rel="nofollow noopener">clone your GitHub repository locally</a> using the <code>git clone</code> command? (The instructions you reference are for starting a brand new Pages site locally, so they’re not really designed for your situation.) When you clone your GitHub repository locally, then it should be all up to date and you don’t need to checkout master or fetch or anything. You can go straight to installing Jekyll and generating the site files.</p>

**Comment by berndbausch** [comment]
<p>Thanks, this is probably the answer I am looking for. Note, though, that the referenced page doesn’t say that the instructions are for brandnew sites only. It starts as follows:</p>
<p><em>You can set up a local version of your Jekyll GitHub Pages site to test changes to your site locally. We highly recommend installing Jekyll to preview your site and help troubleshoot failed Jekyll builds.</em></p>
<p>and later:</p>
<p><em>You may already have Jekyll site files on your local computer if you cloned a Jekyll site repository</em></p>
<p>Meanwhile, I fixed my problem by creating a new site and copying my files into it.</p>
